packages/twenty-front/src/modules/workspace/components/WorkspaceProviderEffect.tsx

```diff
@@ -0,0 +1,46 @@
+import { useRecoilValue, useSetRecoilState } from 'recoil';
+
+import { useGetPublicWorkspaceDataBySubdomainQuery } from '~/generated/graphql';
+
+import { workspacePublicDataState } from '@/auth/states/workspacePublicDataState';
+import { authProvidersState } from '@/client-config/states/authProvidersState';
+import { useEffect } from 'react';
+import { isDefined } from '~/utils/isDefined';
+
+export const WorkspaceProviderEffect = () => {
+  const workspacePublicData = useRecoilValue(workspacePublicDataState);
+
+  const setAuthProviders = useSetRecoilState(authProvidersState);
+  const setWorkspacePublicDataState = useSetRecoilState(
+    workspacePublicDataState,
+  );
+
+  useGetPublicWorkspaceDataBySubdomainQuery({
+    onCompleted: (data) => {
+      setAuthProviders(data.getPublicWorkspaceDataBySubdomain.authProviders);
+      setWorkspacePublicDataState(data.getPublicWorkspaceDataBySubdomain);
+    },
+    onError: (err) => {
+      // eslint-disable-next-line no-console
+      console.error(err);
+    },
+  });
+
+  useEffect(() => {
+    try {
+      if (isDefined(workspacePublicData?.logo)) {
```

Comment:
weird to do that in a useEffect and manipulating the DOM, can't we edit the head direclty?